### PR TITLE
 testing button component

### PR DESCRIPTION
**Pull request by MarcoSantarelli**
**Related tasks:**
was added:
* button test

**Related cards:**
* FID-32

**Before testing:**
1. Clone the PR 
2. run the angular project
3. run the command "ng test"
---

### CODE REVIEW

File: src/app/atoms/button/che-button.component.spec.ts [hunk 1]
```diff
@@ -1,25 +1,272 @@
...
+        let component: any;
+        let fixture: ComponentFixture<any>;
+
+        beforeEach(() => {
+            // Arrange
+            fixture = createComponent(CheButtonF6Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+        it('should have attribute size with value "sm"', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts:168]: Add another test case to test the button size change.

File: src/app/atoms/button/che-button.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,272 @@
...
+        let component: any;
+        let fixture: ComponentFixture<any>;
+
+        beforeEach(() => {
+            // Arrange
+            fixture = createComponent(CheButtonF2Component);
+            component = fixture.componentInstance;
+            fixture.detectChanges();
+        });
+        describe('Has color', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts:95]: If it has a color

File: src/app/atoms/button/che-button.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,272 @@
...
+
+                // Assert
+                expect(cheButton.nativeElement.getAttribute('color')).toEqual('warning');
+                expect(button.nativeElement.classList.contains('btn-outline-warning')).toBeTruthy('Shold to have the class btn-outline-warning');
+
+            });
+        });
+
+        // fixture 3
+        describe('Has no color', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If it has no color

File: src/app/atoms/button/che-button.component.spec.ts [hunk 4]
```diff
@@ -1,25 +1,272 @@
...
+            const style = window.getComputedStyle(button.nativeElement, ':before');
+
+            // Act
+
+            // Assert
+            expect(button.nativeElement.classList.contains('btn-outline-warning')).toBeTruthy('Shold to have the class btn-outline-warning');
+            expect(style.getPropertyValue('background-color')).toEqual('rgba(0, 0, 0, 0)');
+        });
+        // fixture 1
+        describe('Has color', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If it has a color

File: src/app/atoms/button/che-button.component.spec.ts [hunk 5]
```diff
@@ -1,25 +1,272 @@
...
+                // Assert
+                expect(cheButton.nativeElement.getAttribute('color')).toEqual('primary');
+                expect(button.nativeElement.classList.contains('btn-primary')).toBeTruthy('Shold to have the class btn-primary');
+
+
+            });
+        });
+
+        // fixture 4
+        describe('Has no color', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If it has no color

File: src/app/atoms/button/che-button.component.spec.ts [hunk 6]
```diff
@@ -1,25 +1,272 @@
...
+            });
+
+
+        });
+
+    });
+
+
+    // fixture 5
+    describe('If a button is disabled', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If the button is disabled

File: src/app/atoms/button/che-button.component.spec.ts [hunk 7]
```diff
@@ -1,25 +1,272 @@
...
-import { CheButtonComponent } from './che-button.component';
+import {CheButtonComponent} from './che-button.component';
+import {Component, Provider, Type} from '@angular/core';
+import {FormsModule} from '@angular/forms';
+import {By} from '@angular/platform-browser';
+import * as $ from 'jquery';
+
+describe('che-button Component', () => {
+    // fixture 1
+    describe('If  button is outline', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If the button is outline

File: src/app/atoms/button/che-button.component.spec.ts [hunk 8]
```diff
@@ -1,25 +1,272 @@
...
+            // Act
+            const button = fixture.debugElement.query(By.css('button'))!;
+            expect(button.nativeElement.getAttribute('disabled')).toBeDefined('should be defined the attribute disabled');
+        });
+
+
+    });
+
+    // fixture 6
+    describe('If a button with size', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If it has a size

File: src/app/atoms/button/che-button.component.spec.ts [hunk 9]
```diff
@@ -1,25 +1,272 @@
...
+            // Act
+            const button = fixture.debugElement.query(By.css('button'))!;
+            expect(button.nativeElement.classList.contains('btn-sm')).toBeTruthy('Should to have the class btn-sm');
+        });
+
+
+    });
+
+
+    describe('If is a button without size ', () => {
```
Comment (danabalon) [inline src/app/atoms/button/che-button.component.spec.ts]: If it has no size

**Review by danabalon — APPROVED** [review]
Approved  with a couple of comments.

File: src/app/atoms/button/che-button.component.spec.ts [hunk 10]
```diff
@@ -1,25 +1,298 @@
-import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
```
Comment (aderenzis) [inline src/app/atoms/button/che-button.component.spec.ts:1]: unused async

File: src/app/atoms/button/che-button.component.spec.ts [hunk 11]
```diff
@@ -1,25 +1,298 @@
...
 
-import { CheButtonComponent } from './che-button.component';
+import {CheButtonComponent} from './che-button.component';
+import {Component, Provider, Type} from '@angular/core';
+import {FormsModule} from '@angular/forms';
+import {By} from '@angular/platform-browser';
+import * as $ from 'jquery';
```
Comment (aderenzis) [inline src/app/atoms/button/che-button.component.spec.ts:7]: unused jquery

File: src/app/atoms/input/che-input.component.spec.ts [hunk 12]
```diff
@@ -1,25 +1,315 @@
...
+    //
+    //     expect(input.nativeElement.value)
+    //       .toBe('a value', 'should to have the text');
+    //
+    //     expect(divGroup.nativeElement.classList.contains('has-danger'))
+    //       .toBe(false, 'should not to have the class has-danger');
+    //
+    //   });
+    // });
+
```
Comment (aderenzis) [inline src/app/atoms/input/che-input.component.spec.ts:181]: Commented code

**Comment by aderenzis** [comment]
The PR description only mentions changes over che-button, however, several components were modified. It's important to check all file change before creating the pull request and write its description according to the changes.

**Review by aderenzis — APPROVED** [review]
The PR description only mentions changes over che-button, however, several components were modified. It's important to check all file change before creating the pull request and write its description according to the changes.

**Pull request merged**